action/candidate_update.go

```diff
@@ -200,9 +200,15 @@ func (cu *CandidateUpdate) EncodeABIBinary() ([]byte, error) {
 }
 
 func (cu *CandidateUpdate) encodeABIBinary() ([]byte, error) {
-	operatorEthAddr := common.BytesToAddress(cu.operatorAddress.Bytes())
-	rewardEthAddr := common.BytesToAddress(cu.rewardAddress.Bytes())
-	data, err := _candidateUpdateMethod.Inputs.Pack(cu.name, operatorEthAddr, rewardEthAddr)
+	if cu.operatorAddress == nil {
+		return nil, ErrAddress
+	}
+	if cu.rewardAddress == nil {
+		return nil, ErrAddress
+	}
```

Comment:
missing these 2 checks caused panic

Reply:
yes, that is the native side, it has the null-check

this code is used to decode the txs in the block when returning a block via web3 format, it is missing the null-check

Reply:
ok, then should we return normally using ZeroAddress to represent the nil param, or just return error to damage and compatibility with eth?